examples/seismic/source.py

```diff
@@ -1,60 +1,105 @@
+from scipy import signal
+
 from devito import Dimension
 from devito.function import SparseTimeFunction
 from devito.logger import error
 
+from cached_property import cached_property
+
 import numpy as np
 try:
     import matplotlib.pyplot as plt
 except:
     plt = None
 
-__all__ = ['PointSource', 'Receiver', 'Shot', 'RickerSource', 'GaborSource']
+__all__ = ['PointSource', 'Receiver', 'Shot', 'WaveletSource',
+           'RickerSource', 'GaborSource', 'TimeAxis']
+
+
+class TimeAxis:
+    def __init__(self, start, step, num):
+        try:
+            self.start = start
+            self.stop = start + step*num
+            self.step = step
+            self.num = num
+        except:
+            error("error in time range parameters")
+
+    def __str__(self):
+        return "TimeAxis(%g, %g, %g, %g)" % (self.start, self.stop, self.step, self.num)
+
+    @cached_property
+    def time_values(self):
+        return np.linspace(self.start, self.stop, self.num, endpoint=False)
 
 
 class PointSource(SparseTimeFunction):
     """Symbolic data object for a set of sparse point sources
 
-    :param name: Name of the symbol representing this source
+    :param name: Name of the symbol representing this source.
     :param grid: :class:`Grid` object defining the computational domain.
-    :param coordinates: Point coordinates for this source
-    :param data: (Optional) Data values to initialise point data
-    :param ntime: (Optional) Number of timesteps for which to allocate data
-    :param npoint: (Optional) Number of sparse points represented by this source
-    :param time_order: (Optional) Time discretization order (defaults to 2)
+    :param coordinates: Point coordinates for this source.
+    :param time_range: TimeAxis(start, step, num) object.
```

Comment:
:class:`TimeAxis`(...) is what Sphinx likes

Reply:
Fixed in 3cd5c73a